core/renderers/measurables/rows.js

```diff
@@ -290,8 +290,9 @@ Blockly.utils.object.inherits(Blockly.blockRendering.TopRow,
  * @return {boolean} Whether or not the top row has a left square corner.
  */
 Blockly.blockRendering.TopRow.prototype.hasLeftSquareCorner = function(block) {
-  var hasHat = (block.hat ? block.hat === 'cap' : Blockly.BlockSvg.START_HAT) &&
-    !block.outputConnection && !block.previousConnection;
+  var hasHat = (typeof this.block_.hat !== 'undefined' ?
```

Comment:
block.hat !== 'undefined' instead of this.block_.hat.

Reply:
Done.